src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -133,7 +144,7 @@ public byte[] get(String name, byte[] key) {
 	}
 
 	@Override
-	public byte[] putIfAbsent(String name, byte[] key, byte[] value, @Nullable Duration ttl) {
+	public byte[] putIfAbsent(String name, byte[] key, byte[] value, @Nullable Duration ttl, @Nullable Duration maxIdle) {
```

Comment:
I don't think this signature change makes any sense in any of the changed `RedisCacheWriter` methods: `get(..)`, `put(..)` or `putIfAbsent(..)`; see [below](src/main/java/org/springframework/data/redis/cache/RedisCacheWriter.java).

I'd prefer, if we decide to add any option to configure TTI (perhaps overriding TTL when TTI < TTL) per cache entry by key at all, then the configuration would be included in the `RedisCacheConfiguration` per cache (not unlike TTL) and the `RedisCacheConfiguration` object is passed to the components of the cache implementation as needed to alter the caching behavior. Otherwise with each new option, the signature will only grow.